File: services/block-e-chunking/fixtures/code/python/string_utils.py
import re
from typing import List

class StringUtils:
    """String manipulation utilities."""
    
    @staticmethod
    def camel_to_snake(name: str) -> str:
        """Convert CamelCase to snake_case."""
        s1 = re.sub('(.)([A-Z][a-z]+)', r'\1_\2', name)
        return re.sub('([a-z0-9])([A-Z])', r'\1_\2', s1).lower()
    
    @staticmethod
    def snake_to_camel(name: str) -> str:
        """Convert snake_case to CamelCase."""
        components = name.split('_')
        return ''.join(x.title() for x in components)
    
    @staticmethod
    def truncate(text: str, max_length: int, suffix: str = '...') -> str:
        """Truncate text to max length."""
        if len(text) <= max_length:
            return text
        return text[:max_length - len(suffix)] + suffix
    
    @staticmethod
    def split_into_chunks(text: str, chunk_size: int) -> List[str]:
        """Split text into chunks of specified size."""
        return [text[i:i + chunk_size] for i in range(0, len(text), chunk_size)]

File: fixtures/code_corpus/go/string_utils.go
package main

import (
	"fmt"
	"strings"
	"unicode"
)

type StringUtils struct{}

func NewStringUtils() *StringUtils {
	return &StringUtils{}
}

func (s *StringUtils) CamelToSnake(name string) string {
	var result []rune
	for i, r := range name {
		if unicode.IsUpper(r) && i > 0 {
			result = append(result, '_')
		}
		result = append(result, unicode.ToLower(r))
	}
	return string(result)
}

func (s *StringUtils) SnakeToCamel(name string) string {
	parts := strings.Split(name, "_")
	for i, part := range parts {
		if len(part) > 0 {
			parts[i] = strings.Title(part)
		}
	}
	return strings.Join(parts, "")
}

func (s *StringUtils) Truncate(text string, maxLen int, suffix string) string {
	if len(text) <= maxLen {
		return text
	}
	return text[:maxLen-len(suffix)] + suffix
}

func (s *StringUtils) SplitIntoChunks(text string, chunkSize int) []string {
	var chunks []string
	for i := 0; i < len(text); i += chunkSize {
		end := i + chunkSize
		if end > len(text) {
			end = len(text)
		}
		chunks = append(chunks, text[i:end])
	}
	return chunks
}

func main() {
	utils := NewStringUtils()
	fmt.Println(utils.CamelToSnake("CamelCase"))
	fmt.Println(utils.SnakeToCamel("snake_case"))
	fmt.Println(utils.Truncate("Hello World", 5, "..."))
}
